analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/InvocationCheck.cs

```diff
@@ -0,0 +1,76 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
+{
+    internal class InvocationCheck : SymbolicCheck
+    {
+        public override ProgramState[] PreProcess(SymbolicContext context) =>
+            context.Operation.Instance.Kind == OperationKindEx.Invocation
+                && PreProcessInvocation(context.State, IInvocationOperationWrapper.FromOperation(context.Operation.Instance)) is { } newState
+                    ? newState
+                    : base.PreProcess(context);
+
+        private ProgramState[] PreProcessInvocation(ProgramState state, IInvocationOperationWrapper invocation)
+        {
+            if (ArgumentOfIsNullOrEmpty(invocation) is { } symbol)
+            {
+                if (state[symbol] is { } symbolicValue && symbolicValue.HasConstraint(ObjectConstraint.Null))
+                {
+                    // IsNullOrEmpty will always return "true". For the NotNull case, we can not tell for sure (string could be empty)
+                    return new[] { SetOperationConstraint(BoolConstraint.True) };
+                }
+
+                return new[]
+                {
+                    SetOperationConstraint(BoolConstraint.True).SetSymbolConstraint(symbol, ObjectConstraint.Null),
+                    SetOperationConstraint(BoolConstraint.True).SetSymbolConstraint(symbol, ObjectConstraint.NotNull),
+                    SetOperationConstraint(BoolConstraint.False).SetSymbolConstraint(symbol, ObjectConstraint.NotNull),
+                };
+            }
+
+            return default;
+
+            ProgramState SetOperationConstraint(BoolConstraint boolConstraint) =>
+                state.SetOperationConstraint(invocation.WrappedOperation, boolConstraint);
+        }
+
+        private static ISymbol ArgumentOfIsNullOrEmpty(IInvocationOperationWrapper invocation) =>
+            invocation is
+            {
+                TargetMethod:
+                {
+                    IsStatic: true,
+                    ContainingType.SpecialType: SpecialType.System_String,
+                    Name: nameof(string.IsNullOrEmpty) or nameof(string.IsNullOrWhiteSpace),
+                    Parameters: { Length: 1 } parameters
+                },
+                Arguments: { Length: 1 } arguments,
+            }
+            && parameters[0] is { Name: "value", Type.SpecialType: SpecialType.System_String }
```

Comment:
We shouldn't need this part after addresing other comments.. The parameter check should not be needed. 
```suggestion
```

Reply:
I kept the argument length check (needed for safely indexing into it) and the "IsStatic" check.

Reply:
By design, we cannot have `IsNullOrXxxx` as an instance method => it must be static.
Having it static, it must have at least one argument. There's no point for .NET to add method without.

We can safely remove those checks